lib/settings_yaml.rb

```diff
@@ -19,40 +19,75 @@
 #       enabled: true
 #
 # @example `config/initializers/00_settings.rb`
-#   require 'handsome_extensions/settings'
+#   require 'settings_yaml'
 #
-#   Settings = HandsomeExtensions::Settings.new.load!
+#   Settings = SettingsYaml.load!
 #
 # @example Accessing settings
 #   if Settings.cloud.enabled
 #     ...
 #   end
-class SettingsYaml < Hashie::Mash
-  # @return [Hasie::Mash]
-  def load!
-    self.deep_merge! settings_from_file('settings.yml')
-    self.deep_merge! settings_from_file('settings.local.yml')
-    self
-  end
+class SettingsYaml
+
+  APPLICATION_SETTINGS_ROOT_NAME = 'application'
+
+  class << self
+    # @param filename [String]
+    # @param environment [String]
+    # @return [Hasie::Mash]
+    def load!(filename='settings', environment=nil)
+      main_settings, local_settings  = FileParser.parse("#{filename}.yml"), FileParser.parse("#{filename}.local.yml")
```

Comment:
Two separate lines would probably look better. At lease they will fit in 1 line without wrapping.